sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -895,14 +895,134 @@ def test_non_euclidean_kneighbors():
     # Raise error when wrong parameters are supplied,
     X_nbrs = neighbors.NearestNeighbors(3, metric='manhattan')
     X_nbrs.fit(X)
-    assert_raises(ValueError, neighbors.kneighbors_graph, X_nbrs, 3,
+    assert_raises(ValueError, neighbors.kneighbors_graph, None, 3,
                   metric='euclidean')
     X_nbrs = neighbors.NearestNeighbors(radius=radius, metric='manhattan')
     X_nbrs.fit(X)
-    assert_raises(ValueError, neighbors.radius_neighbors_graph, X_nbrs,
+    assert_raises(ValueError, neighbors.radius_neighbors_graph, None,
                   radius, metric='euclidean')
 
 
+def check_object_arrays(nparray, list_check):
+    for ind, ele in enumerate(nparray):
+        assert_array_equal(ele, list_check[ind])
+
+
+def test_k_and_radius_neighbors_train_is_not_query():
+    """Test kneighbors et.al when query is not training data"""
+
+    for algorithm in ALGORITHMS:
+
+        nn = neighbors.NearestNeighbors(n_neighbors=1, algorithm=algorithm)
+
+        X = [[0], [1]]
+        nn.fit(X)
+        test_data = [[2], [1]]
+        assert_array_equal(
```

Comment:
But what if someone adds something to these methods, so that they don't have fixed relationships anymore tomorrow? I'll put these in separate functions, but I think they should be there.